CONTRIBUTING.md

```diff
@@ -0,0 +1,159 @@
+
+[theia-ide]
+```
+$ cd /home/theia
+$ mkdir -p /tmp/theiadev_projects
+$ export CHE_PROJECTS_ROOT=/tmp/theiadev_projects
+$ export THEIA_PLUGIN_ENDPOINT_DISCOVERY_PORT='2504'
+$ node src-gen/backend/main.js /tmp/theiadev_projects --hostname=0.0.0.0 --port=3130
+```
+
+Click on `theia-ide/theia-dev` tree node in `MY WORKSPACE` view to open a new browser window with a separete Che-Theia instance.
+To stop the command execution, focus the command output view and click `Ctrl+C`.
+
+## Eclipse Che Sidecar container for Kubernetes plugin
+
+On startup, the repository [che-sidecar-kubernetes-tooling](https://github.com/che-dockerfiles/che-sidecar-kubernetes-tooling) will be cloned to your workspace. The project describes a docker image to run Kubernetes tooling extension.
+You can easily build the image inside `buildah-dev` container and push the omage to your account on [quay.io](quay.io).
+
+### Login to quay.io
+
+The first you need to login to [quay.io](quay.io) repository. For that you can use `'3.1 Login to quay.io'` command in `MY WORKSPACE` view.
+It will ask for the user name, will save it to a temporary file and then will login to [quay.io](quay.io).
+Storing to the file is needed to avoid asking the user name when building and when pushing your image. 
+
+The command will run the following in `buildah-dev` container.
+
+[buildah-dev]
+```
+$ cd /projects/che-sidecar-kubernetes-tooling
+$ read -p "Username: " USERNAME
+$ echo $USERNAME > /tmp/username
+$ buildah login --username $USERNAME quay.io
+```
+
+### Build the image
+
+To build the image use `'3.2 Build the image'`. It will switch to `/projects/che-sidecar-kubernetes-tooling` directory and build the image using a Dockerfile. When finish, the list of available images will be shown.
+
+[buildah-dev]
+```
+$ cd /projects/che-sidecar-kubernetes-tooling
+$ USERNAME=`cat /tmp/username`
+$ buildah bud -t quay.io/$USERNAME/che-sidecar-kubernetes-tooling:dev .
+$ buildah images
+```
+
+### Push to quay.io
+
+Use `'3.3 Push to quay.io'` command to push the image to your account on [quay.io](quay.io). The command will run the following sequence of commands in `buildah-dev` container.
+
+[buildah-dev]
+```
+$ cd /projects/che-sidecar-kubernetes-tooling
+$ USERNAME=`cat /tmp/username`
+$ PLUGIN_IMAGE_ID=`buildah images -q quay.io/$USERNAME/che-sidecar-kubernetes-tooling`
+$ PLUGIN_IMAGE=quay.io/$USERNAME/che-sidecar-kubernetes-tooling:dev
+$ buildah push $PLUGIN_IMAGE_ID docker://$PLUGIN_IMAGE
+```
+
+The memory limitation for `buildah-dev` container is set to 3 gigabytes, which should be enough for mostly cases. In a case you cannot build the image due to lack of the memory, you can easily add more by editing the devfile in the dashboard with the following restarting the workspace.
```

Comment:
for _most_ cases not mostly.

_In case_ not In a case

_lack of memory_ not lack of the memory

"with the following restarting the workspace" needs rephrasing.

Reply:
done